cookieninja/prompt.py

```diff
@@ -186,57 +198,53 @@ def prompt_for_config(context, no_input=False):
     # These must be done first because the dictionaries keys and
     # values might refer to them.
     for key, raw in context['cookiecutter'].items():
+        try:
+            rendered_raw = render_variable(env, raw, cookiecutter_dict)
+        except UndefinedError as err:
+            msg = f"Unable to render variable '{key}'"
+            raise UndefinedVariableInTemplate(msg, err, context) from err
+
         if key.startswith('_') and not key.startswith('__'):
             cookiecutter_dict[key] = raw
             continue
         elif key.startswith('__'):
-            cookiecutter_dict[key] = render_variable(env, raw, cookiecutter_dict)
+            cookiecutter_dict[key] = rendered_raw
             continue
 
-        try:
-            if isinstance(raw, list):
-                # We are dealing with a choice variable
-                val = prompt_choice_for_config(
-                    cookiecutter_dict, env, key, raw, no_input
-                )
-                cookiecutter_dict[key] = val
-            elif isinstance(raw, bool):
-                # We are dealing with a boolean variable
-                if no_input:
-                    cookiecutter_dict[key] = render_variable(
-                        env, raw, cookiecutter_dict
-                    )
-                else:
-                    cookiecutter_dict[key] = read_user_yes_no(key, raw)
-            elif not isinstance(raw, dict):
-                # We are dealing with a regular variable
-                val = render_variable(env, raw, cookiecutter_dict)
-
-                if not no_input:
-                    val = read_user_variable(key, val)
-
-                cookiecutter_dict[key] = val
-        except UndefinedError as err:
-            msg = f"Unable to render variable '{key}'"
-            raise UndefinedVariableInTemplate(msg, err, context) from err
+        raw = rendered_raw
```

Comment:
No need for this variable, as it's the same as `rendered_raw`, but used interchangeably in the if statements.
We can just use `rendered_raw` from here on.

Reply:
Fixed